src/phylum/ci/ci_none.py

```diff
@@ -0,0 +1,138 @@
+    return remote
+
+
+class CINone(CIBase):
+    """Provide methods for operating outside of a known CI environment."""
+
+    def __init__(self, args: argparse.Namespace) -> None:
+        self.ci_platform_name = "No CI"
+        super().__init__(args)
+
+    def check_prerequisites(self) -> Any:
+        """Ensure the necessary pre-requisites are met and bail when they aren't.
+
+        These are the current pre-requisites for when no CI environments/platforms is detected:
+          * Have `git` installed and available for use on the PATH
+          * Run the script from the root of a git repository
+        """
+        with super().check_prerequisites():
+            if shutil.which("git"):
+                print(" [+] `git` binary found on the PATH")
+            else:
+                raise SystemExit(" [!] `git` is required to be installed and available on the PATH")
+            git_dir = Path.cwd() / ".git"
+            if git_dir.is_dir():
+                print(" [+] Existing `.git` directory was found at the current working directory")
+            else:
+                raise SystemExit(" [!] This script expects to be run from the top level of a `git` repository")
+
+    @property
+    def phylum_label(self) -> str:
+        """Get a custom label for use when submitting jobs with `phylum analyze`."""
+        cmd = "git branch --show-current".split()
+        current_branch = subprocess.run(cmd, check=True, text=True, capture_output=True).stdout.strip()
+
+        # This is the unique key that git uses to refer to the blob type data object for the lockfile.
+        # Reference: https://git-scm.com/book/en/v2/Git-Internals-Git-Objects
+        cmd = f"git hash-object {self.lockfile}".split()
+        lockfile_hash_object = subprocess.run(cmd, check=True, text=True, capture_output=True).stdout.strip()
+        label = f"{SCRIPT_NAME}_{self.ci_platform_name}_{current_branch}_{lockfile_hash_object}"
+        label = label.replace(" ", "-")
+
+        return label
+
+    def _is_lockfile_changed(self, lockfile: Path) -> bool:
+        """Predicate for detecting if the given lockfile has changed.
+
+        For the case of operating outside of a CI platform, some assumptions are made:
+          * There is only one remote configured for the repository
+          * The diff is comparing against the remote and not another ref
+          * The diff is comparing by using the files at the current HEAD
+
+        The usefulness of this approach is limited in that lockfile changes must already be committed to be detected.
+
+        References:
+        https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-emgitdiffemltoptionsgtltcommitgtltcommitgt--ltpathgt82308203
+        """
+        remote = git_remote()
+        cmd = f"git diff {remote}... -- {lockfile.resolve()}"
+        lockfile_diff = subprocess.run(cmd.split(), check=True, text=True, capture_output=True).stdout
+        return bool(len(lockfile_diff))
```

Comment:
From the `git diff` docs:

```
       --exit-code
           Make the program exit with codes similar to diff(1). That is, it exits with 1 if there were differences and 0 means no differences.
```

Probably cleaner to use that than to check the length of the output.

Reply:
Roger that...changed.